src/arm/windows/init.c

```diff
@@ -0,0 +1,252 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include <cpuinfo.h>
+#include <cpuinfo/internal-api.h>
+#include <cpuinfo/log.h>
+
+#include "windows_arm_init.h"
+
+/* Efficiency class = 0 means little core, while 1 means big core for now */
+#define MAX_WOA_VALID_EFFICIENCY_CLASSES		2
+#define VENDOR_NAME_MAX		CPUINFO_PACKAGE_NAME_MAX
+
+struct cpuinfo_arm_isa cpuinfo_isa;
+
+static void set_cpuinfo_isa_fields();
+static bool get_system_info_from_registry(
+	struct woa_chip_info** chip_info,
+	enum cpu_info_vendor* vendor);
+
+struct vendor_info {
+	char vendor_name[VENDOR_NAME_MAX];
+	enum cpu_info_vendor cpu_info_vendor;
+};
+
+/* Please add new vendor here! */
+static struct vendor_info vendors[] = {
+	{
+		"Qualcomm",
+		cpuinfo_vendor_qualcomm
+	}
+};
+
+/* Please add new SoC/chip info here! */
+static struct woa_chip_info woa_chips[] = {
+	/* Microsoft SQ1 Kryo 495 4 + 4 cores (3 GHz + 1.80 GHz) */
+	"Microsoft SQ1",
+	woa_chip_name_microsoft_sq_1,
+	{
+		{
+			cpuinfo_uarch_cortex_a55,
+			1800000000,
+		},
+		{
+			cpuinfo_uarch_cortex_a76,
+			3000000000,
+		}
+	},
+	/* Microsoft SQ2 Kryo 495 4 + 4 cores (3.15 GHz + 2.42 GHz) */
+	"Microsoft SQ2",
+	woa_chip_name_microsoft_sq_2,
+	{
+		{
+			cpuinfo_uarch_cortex_a55,
+			2420000000,
+		},
+		{
+			cpuinfo_uarch_cortex_a76,
+			3150000000
+		}
+	}
+};
+
+BOOL CALLBACK cpuinfo_arm_windows_init(
+	PINIT_ONCE init_once, PVOID parameter, PVOID* context)
+{
+	struct woa_chip_info *chip_info = NULL;
+	enum cpu_info_vendor vendor = cpuinfo_vendor_unknown;
+	BOOL result = false;

```

Comment:
C99 supports `bool` type, isn't it?
```suggestion
	bool result = false;

```